### PR TITLE
Update SP connections config

### PR DESCRIPTION
**Pull request by sergioc**

### CODE REVIEW

File: servicepulse/host-config.md [hunk 1]
```diff
@@ -24,13 +24,17 @@ The default value for ImagePath is:
...
- * Go to the installation directory for ServicePulse. Typically this is `C:\Program Files (x86)\Particular Software\ServicePulse\`
- * Go to the `app\js` directory and edit `app.constants.js`
- * Delete any `*.js` files in the `app` directory
- * Change the value of the url after `service_control_url`
\ No newline at end of file
+The default connections can easily be overriden via the Connections configuration screen in ServicePulse:
+
+![Connections configuration](images/connections-configuration.png 'width=500')
+
+To configure connections via the Connections configuration screen, the ServiceControl (Monitoring) URLs need to be entered, with the port number included. This configuration will persist only locally and is read in the following order:
```
Comment (mauroservienti) [inline servicepulse/host-config.md]:
```suggestion
To configure connections via the Connections configuration screen, the ServiceControl and ServiceControl Monitoring full URLs need to be entered, including port number and scheme. Connection configuration is persisted only locally and is read in the following order:
```

File: servicepulse/host-config.md [hunk 2]
```diff
@@ -24,13 +24,17 @@ The default value for ImagePath is:
...
+
+1. **URL in the browser address bar** - allows easily bookmarking and sharing connection details
```
Comment (mauroservienti) [inline servicepulse/host-config.md]:
```suggestion
1. **Query string in the browser address bar** - allows easily bookmarking and sharing of connection details.
```

File: servicepulse/host-config.md [hunk 3]
```diff
@@ -24,13 +24,17 @@ The default value for ImagePath is:
...
+2. **Local storage** - remembers the last used connection if connection parameters are later not found in the URL
```
Comment (mauroservienti) [inline servicepulse/host-config.md]:
```suggestion
2. **Local storage** - remembers the last used connection if connection parameters are later not found in query string.
```

File: servicepulse/host-config.md [hunk 4]
```diff
@@ -24,13 +24,17 @@ The default value for ImagePath is:
...
- * Change the value of the url after `service_control_url`
\ No newline at end of file
+The default connections can easily be overriden via the Connections configuration screen in ServicePulse:
+
+![Connections configuration](images/connections-configuration.png 'width=500')
+
+To configure connections via the Connections configuration screen, the ServiceControl and ServiceControl Monitoring full URLs need to be entered, including port number and scheme. Connection configuration is persisted only locally and is read in the following order:
+
+1. **Query string in the browser address bar** - allows easily bookmarking and sharing of connection details.
+2. **Local storage** - remembers the last used connection if connection parameters are later not found in the URL
```
Comment (HEskandari) [inline servicepulse/host-config.md]: Maybe this?

>If the connection parameters are not found in the URL, the last remembered connection will be used.

File: servicepulse/host-config.md [hunk 5]
```diff
@@ -24,13 +24,17 @@ The default value for ImagePath is:
...
- * Delete any `*.js` files in the `app` directory
- * Change the value of the url after `service_control_url`
\ No newline at end of file
+The default connections can easily be overriden via the Connections configuration screen in ServicePulse:
+
+![Connections configuration](images/connections-configuration.png 'width=500')
+
+To configure connections via the Connections configuration screen, the ServiceControl and ServiceControl Monitoring full URLs need to be entered, including port number and scheme. Connection configuration is persisted only locally and is read in the following order:
+
+1. **Query string in the browser address bar** - allows easily bookmarking and sharing of connection details.
```
[the remark this answers is not in the document]
Comment (sergioc) [reply servicepulse/host-config.md]: QueryString 👍 

Browser's address bar makes it precise.

File: servicepulse/host-config.md (hunk 3 above)
Comment (sergioc) [reply servicepulse/host-config.md]: According to https://en.wikipedia.org/wiki/Query_string a query string goes inside a URL. I modified to make it "query string in URL"

File: servicepulse/host-config.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (sergioc) [reply servicepulse/host-config.md]: Actually, according to https://en.wikipedia.org/wiki/Query_string query string is 2 words, not one.

File: servicepulse/host-config.md (hunk 4 above)
Comment (sergioc) [reply servicepulse/host-config.md]: I turned that suggestion into something slightly better:

> If the connection parameters are not found in the URL, the last successful connection will be used.

File: servicepulse/host-config.md [hunk 6]
```diff
@@ -24,13 +24,17 @@ The default value for ImagePath is:
 Change the value after `--url=` and restart the service.
 
 
-## Changing the ServiceControl URL
+## Connection to ServiceControl and ServiceControl Monitoring
```
Comment (WilliamBZA) [inline servicepulse/host-config.md]:
```suggestion
## Connecting to ServiceControl and ServiceControl Monitoring
```

File: servicepulse/host-config.md [hunk 7]
```diff
@@ -24,13 +24,17 @@ The default value for ImagePath is:
...
 
-ServicePulse is configured to connect to the ServiceControl REST API. To specify the URL to connect do  the following
+ServicePulse will, by default, attempt to connect to ServiceControl and ServiceControl Monitoring using the URLs `http://localhost:33333/api/` and `http://localhost:33633/`, respectively.
```
Comment (WilliamBZA) [inline servicepulse/host-config.md]:
```suggestion
ServicePulse will by default, attempt to connect to ServiceControl and ServiceControl Monitoring using the URLs `http://localhost:33333/api/` and `http://localhost:33633/`, respectively.
```

**Comment by WilliamBZA** [comment]
We need to update all the links in the tutorials:

```
The link /servicepulse/host-config#changing-the-servicecontrol-url doesn't exist, but was referenced in page \samples\servicecontrol\adapter-sqs\sample.md
The link /servicepulse/host-config#changing-the-servicecontrol-url doesn't exist, but was referenced in page \samples\servicecontrol\adapter-sqlserver-multi-schema\sample.md
The link /servicepulse/host-config#changing-the-servicecontrol-url doesn't exist, but was referenced in page \samples\servicecontrol\adapter-sqlserver-multi-schema\sample.md
The link /servicepulse/host-config#changing-the-servicecontrol-url doesn't exist, but was referenced in page \samples\servicecontrol\adapter-sqlserver-multi-instance\sample.md
The link /servicepulse/host-config#changing-the-servicecontrol-url doesn't exist, but was referenced in page \samples\servicecontrol\adapter-rabbitmq-different-topologies\sample.md
The link /servicepulse/host-config#changing-the-servicecontrol-url doesn't exist, but was referenced in page \samples\servicecontrol\adapter-rabbitmq-different-topologies\sample.md
The link /servicepulse/host-config#changing-the-servicecontrol-url doesn't exist, but was referenced in page \samples\servicecontrol\adapter-mixed-transports\sample.md
The link /servicepulse/host-config#changing-the-servicecontrol-url doesn't exist, but was referenced in page \samples\servicecontrol\adapter-mixed-transports\sample.md
The link /servicepulse/host-config#changing-the-servicecontrol-url doesn't exist, but was referenced in page \samples\servicecontrol\adapter-asq-multi-storage-account\sample.md
The link /servicepulse/host-config#changing-the-servicecontrol-url doesn't exist, but was referenced in page \samples\servicecontrol\adapter-asq-multi-storage-account\sample.md
The link /servicepulse/host-config#changing-the-servicecontrol-url doesn't exist, but was referenced in page \samples\servicecontrol\adapter-asb-multi-namespace\sample.md
The link /servicepulse/host-config#changing-the-servicecontrol-url doesn't exist, but was referenced in page \samples\servicecontrol\adapter-asb-multi-namespace\sample.md
```

Putting this comment here as a marker

**Comment by sergioc** [comment]
@WilliamBZA that should be addressed now.

**Comment by sergioc** [comment]
I wonder why the build started failing after https://github.com/Particular/docs.particular.net/pull/4343/commits/431c2b1a6cfab68487838fadb3854f540e72230b

**Comment by WilliamBZA** [comment]
Missing Snippet, `MsmqConfig`

**Comment by HEskandari** [comment]
@sergioc @WilliamBZA The build is fixed 🎉

File: servicecontrol/setting-custom-hostname.md [hunk 8]
```diff
@@ -23,6 +23,4 @@ ServiceControl Management will then validate the settings changes and restart th
 
 ### Updating ServicePulse Configuration to ServiceControl Custom Hostname
 
- 1. Update the ServicePulse configuration file to access the updated ServiceControl hostname & port number. By default, the ServicePulse configuration file is located in `[Program Files]\Particular Software\ServicePulse\app\app.constants.js`.
- 1. Update the value of the `service_control_url` parameter to the specified ServiceControl hostname and IP port number.
- 1. When next accessing ServicePulse, make sure to refresh the browser cache to allow ServicePulse to access the updated configuration settings.
+Refer to the [ServicePulse documention](/servicepulse/host-config.md#configuring-connections-via-the-servicepulse-ui) for guidance on how to change connections configuration in ServicePulse.
```
Comment (sergioc) [inline servicecontrol/setting-custom-hostname.md]: Refer to the [ServicePulse documention](/servicepulse/host-config.md#configuring-connections-via-the-servicepulse-ui) for guidance on how to configure ServicePulse to connect to ServiceControl.

Comment (mauroservienti) [reply servicecontrol/setting-custom-hostname.md]: Thanks @sergioc, incorporated.

**Comment by mauroservienti** [comment]
Marked as WIP to prevent accidental merge

**Comment by mauroservienti** [comment]
rebased on `master` and force-pushed, many conflicts, I'm not sure I merged everything correctly, so better to give this another review.

File: servicepulse/host-config.md [hunk 9]
```diff
@@ -24,13 +24,17 @@ The default value for ImagePath is:
...
- * Delete any `*.js` files in the `app` directory
- * Change the value of the url after `service_control_url`
\ No newline at end of file
+The default connections can easily be overriden via the Connections configuration screen in ServicePulse:
+
+![Connections configuration](images/connections-configuration.png 'width=500')
+
+When entering the connection URL to ServiceControl and ServiceControl Monitoring, the full URLs including the port number and scheme needs to be entered in the Connections screen. Connection configuration is persisted only locally and is read in the following order:
+
+1. **Query string in browser address bar URL** - allows easily bookmarking and sharing of connection details.
```
Comment (adamralph) [inline servicepulse/host-config.md]: @mauroservienti @HEskandari @WilliamBZA I was led here while reviewing https://github.com/Particular/docs.particular.net/pull/4394. I'm not sure I understand this. Does this mean that, when I switch to another ServiceControl instance, the address to ServiceControl is then embedded in the URL for ServicePulse?

Comment (mauroservienti) [reply servicepulse/host-config.md]: Correct.

**Pull request merged**